docs/pages/management/dynamic-configuration/spacelift.mdx

```diff
@@ -0,0 +1,298 @@
+necessary permissions for Terraform to manage resources in your cluster.
+
+On your workstation, _outside_ the git repository you connected to Spacelift,
+add the following content to a file called `spacelift.yaml`:
+
+```yaml
+kind: role
+metadata:
+  name: spacelift
+spec:
+  allow:
+    rules:
+      - resources:
+        - role
+        - user
+        verbs: ['list','create','read','update','delete']
+version: v6
+---
+kind: user
+metadata:
+  name: spacelift
+spec:
+  roles: ['spacelift']
+version: v2
+```
+
+This is a minimal version of the configuration you will need to provide to
+Spacelift, and is only permitted to manage Teleport roles. 
+
+Create the `spacelift` user and role.
+
+```code
+$ tctl create spacelift.yaml
+role 'spacelift' has been created
+user "spacelift" has been created
+```
+
+### Enable impersonation
+
+The `spacelift` user cannot log in to Teleport to retreive credentials, so
+another user must **impersonate** this user in order to request credentials on
+`spacelift`'s behalf.
+
+Create a role that enables your user to impersonate the Terraform user. Paste
+the following YAML document into a file called `spacelift-impersonator.yaml`:
+
+```yaml
+kind: role
+version: v6
+metadata:
+  name: spacelift-impersonator
+spec:
+  allow:
+    impersonate:
+      users: ['spacelift']
+      roles: ['spacelift']
+```
+
+Next, create the role:
+
+```code
+$ tctl create spacelift-impersonator.yaml
+```
+
+(!docs/pages/includes/add-role-to-user.mdx role="spacelift-impersonator"!)
+
+### Export an identity file
+
+(!docs/pages/includes/plugins/identity-export.mdx user="spacelift" client="Spacelift"!)
+
+### Provide an identity file to Spacelift
+
+Click **Stacks > Teleport**. Click the **Environment** tab, then **Edit**. Set
+the dropdown menu that follows to **Mounted file**, and the path to
+`/mnt/workspace/auth.pem`.
+
+Click **Upload file** and select the file called `auth.pem` that you exported
+earlier.
+
+Make sure you click **Secret** next to the entry for the mounted file:
+
+![Uploading the identity file](../../../img/management/spacelift/id-file.png)
+
+## Step 4/4. Declare configuration resources
+
+In your clone of the GitHub repository you chose in order to follow this guide,
+check out a branch from your main branch and add the following to `main.tf`:
+
+```text
+resource "teleport_role" "terraform_test" {
+  metadata = {
+    name        = "terraform-test"
+    description = "Terraform test role"
+    labels = {
+      example = "yes"
+    }
+  }
+}
+
+resource "teleport_user" "terraform-test" {
+  metadata = {
+    name        = "terraform-test"
+    description = "Terraform test user"
+
+    labels = {
+      test = "true"
+    }
+  }
+
+  spec = {
+    roles = [teleport_role.terraform_test.id]
+  }
+}
+```
+
+Commit your changes and open a pull request against your repository. (Do not
+merge it just yet.)
+
+Click **Stacks > Teleport > PRs**, then click the name of the PR you opened.
```

Comment:
Since we're switching context from GitHub to Spacelift:

```suggestion
In the Spacelift UI Click **Stacks > Teleport > PRs**, then click the name of
the PR you opened.
```